spring-statemachine-core/src/main/java/org/springframework/statemachine/access/StateMachineFunction.java

```diff
@@ -24,6 +24,7 @@
  * @param <I> the function type
  * @see StateMachineAccessor
  */
+@Deprecated
```

Comment:
See my comment in `StateMachineAccessor` as deprecation depends on that. If we can't in a sane way support existing use of `StateMachineFunction`, then we could just drop this interface.